sklearn/utils/estimator_checks.py

```diff
@@ -1460,6 +1472,67 @@ def check_estimators_dtypes(name, estimator_orig, strict_mode=True):
                 getattr(estimator, method)(X_train)
 
 
+def check_transformer_preserve_dtypes(
+    name, transformer_orig, strict_mode=True
+):
```

Comment:
Basically, our default dtype is `np.float64` so we don't consider this case. You will need to skip the test, essentially the part checking the equivalence with 64 bits.

Reply:
I change the test such that we can assume that the first value in `preserves_dtype` is actually the default.
It would be consistent with the way `check_array` is working indeed.